src/runtime/particle-execution-context.ts

```diff
@@ -235,36 +235,49 @@ export class ParticleExecutionContext {
     }];
   }
 
-  private async reloadParticle(id: string) {
-    let resolve: Runnable;
-    const p = new Promise<void>(res => resolve = res);
-    this.pendingLoads.push(p);
-
-    // Get the old particle based on the given id and delete the old particle's cache
-    const oldParticle = this.particles.get(id);
-    delete oldParticle.spec.implBlobUrl;
-
-    // Create a new particle and replace the old one
-    const particle: Particle = await this.createParticleFromSpec(id, oldParticle.spec);
-
-    const handleMap = new Map();
-    const registerList: {proxy: Store, particle: Particle, handle: Handle}[] = [];
-    // Create new handles and disable the handles of the old particles
-    oldParticle.handles.forEach((oldHandle) => {
-      this.createHandle(particle, oldParticle.spec, id, oldHandle.name, oldHandle.storage, handleMap, registerList);
-      if (oldHandle instanceof HandleOld) oldHandle.disable(oldParticle);
+  private async reloadParticle(ids: string[]) {
+    // Delete old particles' caches
+    ids.forEach(id => {
+      const oldParticle = this.particles.get(id);
+      if (oldParticle.spec.implBlobUrl) delete oldParticle.spec.implBlobUrl;
```

Comment:
Unrelated to this CL, but: is it ok to just delete the blob url? Does something else need to be notified that this blob is no longer in use?

Reply:
I was thinking more that we might need to garbage collect blobs when their urls are discarded, but that's outside the scope of this CL.